blocks/bricks/recurrent/base.py

```diff
@@ -227,10 +227,11 @@ def scan_function(*args):
             result = pack(result)
             if return_initial_states:
                 # Undo Subtensor
-                for i in range(len(states_given)):
-                    assert isinstance(result[i].owner.op,
-                                      tensor.subtensor.Subtensor)
-                    result[i] = result[i].owner.inputs[0]
+                for i, info in enumerate(outputs_info):
+                    if info is not None:
```

Comment:
This relies on line 216, right? Can you add a comment that None corresponds to an output and shouldn't be included?